src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6242,15 +6242,26 @@ private ScanTypeFlags ScanType(ParseTypeMode mode, out SyntaxToken lastTokenOfTy
                 }
             }
 
-            if (this.CurrentToken.Kind == SyntaxKind.IdentifierToken)
+            // Handle :: as well for error case of an alias used without a preceding identifier.
+            if (this.CurrentToken.Kind is SyntaxKind.IdentifierToken or SyntaxKind.ColonColonToken)
             {
-                result = this.ScanNamedTypePart(out lastTokenOfType);
-                if (result == ScanTypeFlags.NotType)
+                bool isAlias;
+                if (this.CurrentToken.Kind is SyntaxKind.ColonColonToken)
                 {
-                    return ScanTypeFlags.NotType;
+                    lastTokenOfType = null;
```

Comment:
>lastTokenOfType = null; [](start = 20, length = 23)

This looks suspicious.

Reply:
so it technically isn't, but it's there to appease hte flow checker.  HEre's what happens.  We set that to null there, but then reach this loop:

```c#
                for (bool firstLoop = true; IsDotOrColonColon(); firstLoop = false)
                {
                    if (!firstLoop && isAlias)
                    {
                        isAlias = false;
                    }

                    lastTokenOfType = this.EatToken();
```

`IsDotOrColonColon()` will be true, so we'll go into this look.  And we'll then set `lastTokenOfType` to the `::`.  

I can either do the same if the if-branch, or i can comment as to what will happen. I think i'll do the former.